includes/embeds/class-amp-base-embed-handler.php

```diff
@@ -107,4 +107,21 @@ function ( $attr_name ) {
 		}
 		return wp_array_slice_assoc( $matches, $attribute_names );
 	}
+
+	/**
+	 * Replace the node's parent with itself if the parent is a <p> tag, has no attributes and has no other children.
+	 * This usually happens while `wpautop()` processes the element.
+	 *
+	 * @param DOMElement $node Node.
+	 */
+	protected function maybe_unwrap_p_element( DOMElement $node ) {
+		$parent_element = AMP_DOM_Utils::get_parent_element( $node );
+
+		if ( $parent_element && 'p' === $parent_element->nodeName && false === $parent_element->hasAttributes() ) {
+			$children = $parent_element->getElementsByTagName( '*' );
+			if ( 1 === $children->length ) {
```

Comment:
This won't return the expected NodeList if the child node itself has children (grandchildren). Therefore, maybe something like this is needed given that PHP's `dom` extension doesn't support [`childElementCount`](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/childElementCount):

```suggestion
			$child_element_count = array_sum(
				array_map(
					function ( DOMNode $child ) {
						return $child instanceof DOMElement ? 1 : 0;
					},
					$node->childNodes
				)
			);
			if ( 1 === $child_element_count ) {
```